src/translator/html.cpp

```diff
@@ -47,11 +47,19 @@ size_t countPrefixWhitespaces(string_view const &input) {
   return size;
 }
 
+// Formatters used for exception messages combined with format()
 std::ostream &operator<<(std::ostream &out, HTML::Tag const *tag) {
   if (tag == nullptr) return out << "[nullptr]";
-  out << '<' << tag->name << tag->attributes;
-  if (tag->empty) out << '/';
-  return out << '>';
+  switch (tag->type) {
+    case HTML::Tag::ELEMENT:
+      return out << '<' << tag->name << tag->attributes << '>';
+    case HTML::Tag::VOID_ELEMENT:
+      return out << '<' << tag->name << tag->attributes << "/>";
+    case HTML::Tag::COMMENT:
+      return out << "<!--" << tag->data << "-->";
+    case HTML::Tag::PROCESSING_INSTRUCTION:
+      return out << "<?" << tag->data << "?>";
+  }
```

Comment:
There's a compiler warning here on reaching a point of no return, any way to make it happy?

Reply:
I would have hoped that the compiler was intelligent enough to see that these are all possible enum values that `tag->type` can have. But I can add a `default: return out << "[Unknown tag]";` It's only used for exception messages anyway.